template/funcs.go

```diff
@@ -80,6 +80,30 @@ func envFunc(env []string) func(string) (string, error) {
 	}
 }
 
+// envWithDefaultFunc returns a function which checks the value of an environment variable.
+// Invokers can specify their own environment, which takes precedences over any
+// real environment variables.
+// If an environment variable is not found or its value is empty, the default will be used instead.
+func envWithDefaultFunc(env []string) func(string, string) (string, error) {
+	return func(s string, def string) (string, error) {
+		for _, e := range env {
+			split := strings.SplitN(e, "=", 2)
+			k, v := split[0], split[1]
+			if k == s {
+				if len(v) == 0 {
+					return def, nil
+				}
+				return v, nil
+			}
+		}
+		val := os.Getenv(s)
```

Comment:
I was going to suggest using os.LookupEnv() here to only use the default if the Environment variable was unset, to allow for setting it to empty string.. but the description specifically calls out the behavior of using the default when the env is zero length. 

Was there a use case where this behavior was specifically desired or is it more just by chance? If the latter then I would suggest switching to LookupEnv to allow for the broader use case with empty strings.

Thanks.

Reply:
I don't have a specific use case for when a variable is present and empty vs not present. I'll switch it to use LookupEnv and update the documentation

Reply:
Thanks!